generic_permissions/visibilities.py

```diff
@@ -60,6 +60,46 @@ def get_attribute(self, instance):
         return queryset
 
 
+class VisibilitySerializerMixin:
+    """
+    Mixin for serializers to handle visibility of related fields.
+
+    This mixin ensures that when updating (PATCH) existing relationships,
+    any relationships not included in the request (due to visibility settings)
+    are not unintentionally removed. It does this by adding back the existing relations
+    which are not included in the request.
+
+    This mixin should be used in conjunction with the `VisibilityRelatedFieldMixin`
+    for the relationship field and should be defined after the `ValidatorMixin`
+    to ensure validations are performed first.
+    """
+
+    def validate(self, *args, **kwargs):
+        validated_data = super().validate(*args, **kwargs)
+
+        if not self.instance:
+            return validated_data
+
+        # Incoming patches can have a subset of all relations, so we need to
+        # keep the existing relations which are not included in the request.
+        for key, field in self.fields.items():
+            if (
+                type(field) is not VisibilityManyRelatedField
```

Comment:
You should use `isinstance` here, not a direct type check

```suggestion
                not isinstance(field, VisibilityManyRelatedField)
```

Reply:
Done